sql/2019/04_Media/04_17.sql

```diff
@@ -1,28 +1,28 @@
-#StandardSQL
-/*
-04_17  
-
-66GB
-
-top video players - I have chosen a few based on my knowledge... Perhaps there are more?
-- there are 6400 here WHERE Gzip could make a large difference....
-
-*/
-
-
-SELECT url, 
-      respsize, 
-      ext, 
-      mimetype, 
-      _gzip_save,
-      _TABLE_SUFFIX AS client
-FROM `summary_requests.2019_07_01_*` 
-WHERE (LOWER(url) LIKE "%hls.js" OR  
-      LOWER(url) LIKE "%video.js" OR  
-      LOWER(url) LIKE "%shaka.js" OR  
-      LOWER(url) LIKE "%jwplayer.js" OR  
-      LOWER(url) LIKE "brightcove-player-loader.min.js")
-     
-     
-     
-ORDER BY respsize desc
+#standardSQL
+# 04_17: Video player size
+SELECT
+  percentile,
+  client,
+  player,
+  SUM(COUNT(0)) OVER (PARTITION BY client, player) AS requests,
+  ROUND(APPROX_QUANTILES(respSize, 1000)[OFFSET(percentile * 10)] / 1024, 2) AS kbytes
+FROM (
+  SELECT
+    client,
+    respSize,
+    LOWER(REGEXP_EXTRACT(url, '(?i)(hls|video|shaka|jwplayer|brightcove-player-loader)[(?:\\.min)]?\\.js')) AS player
```

Comment:
Took a look at some of the most popular plugins on wordpress for videos and found another we should add: flowplayer.min.js

Reply:
Good call